gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/FileExporter.java

```diff
@@ -235,14 +236,9 @@ public String exportCgmesZipFile(SweData sweData, Map<String, ByteArrayOutputStr
              ZipOutputStream zipOs = new ZipOutputStream(baos)) {
 
             for (var entry : mapCgmesFiles.entrySet()) {
-                zipOs.putNextEntry(new ZipEntry(entry.getKey()));
-                byte[] bytes = new byte[1024];
-                int length;
-                InputStream is = new ByteArrayInputStream(entry.getValue().toByteArray());
-                while ((length = is.read(bytes)) >= 0) {
-                    zipOs.write(bytes, 0, length);
-                }
-                is.close();
+                final String originalFileName = entry.getKey();
+                zipOs.putNextEntry(new ZipEntry(originalFileName + ZIP));
+                zipOs.write(createInternalZip(entry.getValue().toByteArray(), originalFileName + ".xml"));
```

Comment:
If you extract "*.zip" file extension, why don't you do the same with "*.xml" file extension?